Add Extended Euclidean Algorithm

eskeype opened the pull request:
I saw that there was already a GCD algorithm, but I thought it could be nice to include an extended Euclidean algorithm as I know that some number theoretic & cryptographic algorithms make use of these coefficients. Tell me your thoughts!

Commit:
Added BinomialCoefficient algorithm in NumberTheory directory

Commit:
Updated complexity description for BinomialCoefficient

Commit:
Made suggested changes to BinomialCoefficient.cpp

Commit:
Updated BinomialCoefficient.cpp for ULL, and added test cases

Commit:
Merge remote-tracking branch 'faheel/master'

Commit:
Added ExtendedEuclid.cpp

faheel commented:
There are some changes I made to [BinomialCoefficients.cpp](https://github.com/faheel/Algos/blob/master/NumberTheory/BinomialCoefficient.cpp) related to code style (variable and function names, comment style etc.) which I thought of requesting in the reviews before, but then decided that it would be too time consuming to go over each of them.
Since now the changes have been made, it would be nice if you could take a look at them and implement those in this program too.

P.S. More on the code style can be found in the [C++ coding guidelines](https://github.com/faheel/Algos/blob/master/C%2B%2B_CODING_GUIDELINES.md) and in the [Google style guide](https://google.github.io/styleguide/cppguide.html).

Commit:
Merge remote-tracking branch 'faheel/master'

eskeype commented:
Ok, I'll work on the edits. Thanks!

Commit:
Updated comments and IO in ExtendedEuclid.cpp

eskeype commented:
Ok, I updated the comments and I borrowed the user input function you added to BinomialCoefficient.cpp and added it to ExtendedEuclid.cpp. Let me know if I need anything else

faheel commented on NumberTheory/ExtendedEuclid.cpp:
```diff
@@ -0,0 +1,113 @@
+
+    Time Complexity
+    ---------------
+    O(log10(K)), where K = min(A, B) is the smaller of the input integers A and B
```
Base of the logarithm doesn't matter while writing the complexity, and therefore doesn't need to be mentioned.
Just `O(log(K)), where ...` is sufficient.

faheel commented on NumberTheory/ExtendedEuclid.cpp:
```diff
@@ -0,0 +1,113 @@
+    Return value
+    ------------
+    vector<int> M of length 2 which satisfies the equation above
+
```
Since there will always be exactly 2 values that the function would return, it would be better to return them in either a `tuple<int, int>` or an `array<int, 2>`.

Commit:
Changed vector to array in ExtendedEuclid.cpp and updated complexity comment

Commit:
Updated comments for complexity and <array>

eskeype commented:
I updated the complexity description to no longer include the log10, and I opted for array<int,2> instead of vector<int>. Thanks!

faheel commented:
@eskeype Thank you once again for your contribution :smiley:
